### PR TITLE
RDSv2: add cache

### PR DESCRIPTION
**Pull request by kyessenov**
Fix existing glue for RDS.
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/xds.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 Istio Authors
```
Comment (costinm) [inline pilot/pkg/proxy/envoy/v2/xds.go]: Can you rename this to rds for consistency ( eds goes to eds.go, etc)

Comment (kyessenov) [reply pilot/pkg/proxy/envoy/v2/xds.go]: This can be used for EDS, or LDS, more realistically.

**Comment by kyessenov** [comment]
@costinm PTAL. errors seem unrelated.

**Comment by ldemailly** [comment]
thanks for noticing/catching erroneous vendor/ rollback!

**Pull request merged**